ui/src/components/modals/ProcessModal.tsx

```diff
@@ -4,41 +4,43 @@ import getDataStore from '../../data/DataStore';
 import { PathedData } from '../../data/models/PathedData';
 import { ProcessSpecification } from '../../data/models/Valueflows/Knowledge';
 import { Process } from "../../data/models/Valueflows/Plan";
-import { AgentShape } from '../../types/valueflows';
+import { AgentShape, ProcessShape } from '../../types/valueflows';
+import { assignFields } from '../../utils';
 
-const initialState = {
+const initialState: ProcessShape = {
+  id: '',
+  created: null,
+  basedOn: '',
+  plannedWithin: '',
   name: '',
   finished: false,
   note: '',
   classifiedAs: '',
   inScopeOf: '',
-  basedOn: ''
+  hasBegining: null,
+  hasEnd: null,
+  hasPointInTime: null,
+  due: null
 }
 
 interface Props {
-  processSpecificationPath: string;
+  processState: ProcessShape;
   closeModal: () => void;
-  handleAddNode: (item: PathedData) => void;
+  afterward?: (item: PathedData) => void;
```

Comment:
what scenario would afterwards not be needed?

Reply:
It's always needed to update the on screen DisplayNode, since the form can't handle that.

Reply:
It should probably be made into a required field then.